### PR TITLE
feat: ALERT pre- and post-KF AI PID inference

### PR DESCRIPTION
**Pull request by skuditha**
- Replaces the existing ALERT pre-PID implementation with the new AHDC- and ATOF-based TorchScript models.
- Adds post-KF PID inference for every valid ALERT::ai:projections pair.
- Writes predicted particle IDs and class probabilities to the existing ALERT AI PID banks.

### CODE REVIEW

File: reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java [hunk 1]
```diff
@@ -178,7 +182,8 @@ public boolean init() {
                 "AHDC::preclusters", "AHDC::clusters", "AHDC::track",
                 "AHDC::interclusters", "AHDC::docaclusters", "AHDC::ai:prediction",
                 "AHDC::mc", "AHDC::kftrack",
-                "ALERT::projections", "ALERT::ai:projections", "ALERT::prePID");
+                "ALERT::projections", "ALERT::ai:projections", 
+                "ALERT::ai:prePID", "ALERT::ai:postPID");
```
Comment (mathieuouillon) [inline reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java]: Wrong banks name: look into etc/bankdefs/hipo4/alert.json
"name": "ALERT::ai:prepid", and name": "ALERT::ai:pid"

File: reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java [hunk 2]
```diff
@@ -766,11 +761,105 @@ public boolean processDataEventUser(DataEvent event) {
...
+                        float[] prediction = modelPostPID.prediction(features);
+                        if (prediction != null) {
+                            pidResults.add(new PIDResult(trackid, clusterid, prediction));
+                        }
+                    } catch (TranslateException ex) {
+                        LOGGER.warning(() -> "Exception in ALERTEngine PostPID: " + ex);
+                    }
+                }
+            }
+            rbc.appendPIDBank(event, pidResults);
```
Comment (mathieuouillon) [inline reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java]: appendPIDBank is outside the isFinite(startTime) guard, so a zero-row ALERT::ai:pid is appended to every qualifying event. Consider if (!pidResults.isEmpty()).

**Review by mathieuouillon — COMMENTED** [review]
The old translator did logits.softmax(1) before toFloatArray(). Both new translators (ModelPrePID.java:290, ModelPostPID.java:84) read the raw output tensor. argmax is unaffected, so pid/prepid are fine either way, but if the exported TorchScript modules end in a Linear rather than a Softmax, then p2212/prob_2212/... are logits, and downstream users of a bank field literally named prob_* will be misled. Can you confirm the export includes the softmax?

File: reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java (hunk 2 above)
Comment (skuditha) [reply reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java]: (1) Confirmed. I inspected the TorchScript graphs for both pre-PID models and the post-KF model. All three exported modules apply `softmax(..., -1)` inside their `forward()` method before returning the output tensor. Therefore, the values read by the new translators are already normalized class probabilities, not logits. Applying another softmax in Java would double-transform the probabilities, so the current translators are correct as written.
(2) Thank you! I will fix it. But it does not affect the output for some reason.
(3) Will do. Again, a zero row ALERT::ai:pid did not appear in my tests. So, there seems to be no effect.

**Comment by skuditha** [comment]
Resolved Mathieu's comments. CI will fail until the pull request to merge the models to the networks repository is approved. (https://code.jlab.org/hallb/clas12/coatjava/networks/-/merge_requests/13)

File: reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java (hunk 2 above)
Comment (skuditha) [reply reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java]: For (3) The reason why the zero row banks don't appear is because there is another guard in the recobankwriter. Just to be safe, I pulled the appendPIDBank inside the starttime guard.

**Comment by c-dilks** [comment]
I merged https://code.jlab.org/hallb/clas12/coatjava/networks/-/merge_requests/13 and manually triggered @dependabot to create https://github.com/JeffersonLab/coatjava/pull/1369.
Once that's merged, you can rebase this branch or merge `development` into it.

**Comment by baltzell** [comment]
@skuditha Are you sure you addressed all?  Looks like the bank names may still not exist.  (Also, you can mark them "resolved" if they are so.)

File: reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java (hunk 2 above)
Comment (skuditha) [reply reconstruction/alert/src/main/java/org/jlab/service/alert/ALERTEngine.java]: added additional guard

**Review by baltzell — APPROVED** [review]
